### ISSUE TITLE
the accuracy is always acc=0.000000

### ISSUE THREAD

**Issue by binbinmeng** [deepinsight/insightface#405]
CUDA_VISIBLE_DEVICES='0,1,2,3,4,5,6,7' python -u train_softmax.py --network r100 --loss-type 4 --margin-m 0.5 --data-dir /workspace/dataset/faces_ms1m_112x112  --prefix ../model-r100 --per-batch-size 64
[14:46:53] src/operator/nn/./cudnn/./cudnn_algoreg-inl.h:130: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
INFO:root:Epoch[0] Batch [20]	Speed: 1195.56 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [40]	Speed: 1277.10 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [60]	Speed: 1268.39 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [80]	Speed: 1306.79 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [100]	Speed: 1279.49 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [120]	Speed: 1238.91 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [140]	Speed: 1300.48 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [160]	Speed: 1313.65 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [180]	Speed: 1318.43 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [200]	Speed: 1309.14 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [220]	Speed: 1295.77 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [240]	Speed: 1291.87 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [260]	Speed: 1291.32 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [280]	Speed: 1300.03 samples/sec	acc=0.000000

**Comment by nttstar** [comment]
be more patient :)

**Comment by AmigoCDT** [comment]
If acc is always 0.000, maybe learning rate is too large. Try --lr 0.01 may help

**Comment by Joll123** [comment]
@binbinmeng  I also meet such a problem, have you solved it? Thanks

**Comment by binbinmeng** [comment]
It is not issue, be more patient. I have tested resnet and mobilenet,getting similar results as described.




| |
孟宾宾
邮箱：15221657504@163.com
|

签名由 网易邮箱大师 定制

On 11/02/2018 09:49, Joll123 wrote:

@binbinmeng I also meet such a problem, have you solved it? Thanks

**Comment by clhne** [comment]
> CUDA_VISIBLE_DEVICES='0,1,2,3,4,5,6,7' python -u train_softmax.py --network r100 --loss-type 4 --margin-m 0.5 --data-dir /workspace/dataset/faces_ms1m_112x112 --prefix ../model-r100 --per-batch-size 64
> [14:46:53] src/operator/nn/./cudnn/./cudnn_algoreg-inl.h:130: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
> INFO:root:Epoch[0] Batch [20]	Speed: 1195.56 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [40]	Speed: 1277.10 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [60]	Speed: 1268.39 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [80]	Speed: 1306.79 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [100]	Speed: 1279.49 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [120]	Speed: 1238.91 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [140]	Speed: 1300.48 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [160]	Speed: 1313.65 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [180]	Speed: 1318.43 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [200]	Speed: 1309.14 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [220]	Speed: 1295.77 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [240]	Speed: 1291.87 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [260]	Speed: 1291.32 samples/sec	acc=0.000000
> INFO:root:Epoch[0] Batch [280]	Speed: 1300.03 samples/sec	acc=0.000000

Same to you.
CUDA_VISIBLE_DEVICES='0,1,2,3,4,5,6,7' 
Do you use 8 GPUs ?
and what GPUs you use ?
Thanks.

**Comment by liujianzhao6328057** [comment]
How long did the model take to get the satisfactory result? I choose the MS-Celeb-1M dataset for training, (3,804,846 faces over 85,164 identities) , 5 GPUs ,but the speed seems slow and the. accuracy always be 0.

**Comment by becauseofAI** [comment]
@binbinmeng 
After a long training period, I finally got the following results:
```
2019-07-11 08:38:16,452 Epoch[0] Batch [10980-11000]    Speed: 17.50 samples/sec        acc=0.000586    lossvalue=28.366652
2019-07-11 08:41:49,111 Epoch[0] Batch [11000-11020]    Speed: 24.08 samples/sec        acc=0.001758    lossvalue=28.055955
2019-07-11 08:45:19,096 Epoch[0] Batch [11020-11040]    Speed: 24.38 samples/sec        acc=0.000977    lossvalue=27.956807
2019-07-11 08:48:52,124 Epoch[0] Batch [11040-11060]    Speed: 24.03 samples/sec        acc=0.001563    lossvalue=28.059857
2019-07-11 08:52:18,685 Epoch[0] Batch [11060-11080]    Speed: 24.79 samples/sec        acc=0.001563    lossvalue=27.804412
```
Maybe you need to wait for a very long time patiently.

**Comment by jhgfkdj** [comment]
> @binbinmeng
> After a long training period, I finally got the following results:
> 
> ```
> 2019-07-11 08:38:16,452 Epoch[0] Batch [10980-11000]    Speed: 17.50 samples/sec        acc=0.000586    lossvalue=28.366652
> 2019-07-11 08:41:49,111 Epoch[0] Batch [11000-11020]    Speed: 24.08 samples/sec        acc=0.001758    lossvalue=28.055955
> 2019-07-11 08:45:19,096 Epoch[0] Batch [11020-11040]    Speed: 24.38 samples/sec        acc=0.000977    lossvalue=27.956807
> 2019-07-11 08:48:52,124 Epoch[0] Batch [11040-11060]    Speed: 24.03 samples/sec        acc=0.001563    lossvalue=28.059857
> 2019-07-11 08:52:18,685 Epoch[0] Batch [11060-11080]    Speed: 24.79 samples/sec        acc=0.001563    lossvalue=27.804412
> ```
> 
> Maybe you need to wait for a very long time patiently.

how do you print the lossvalue?
